lib/matplotlib/rcsetup.py

```diff
@@ -718,6 +719,54 @@ def visit_Attribute(self, node):
         self.generic_visit(node)
 
 
+def validate_legend_loc(loc):
+    """
+    Confirm that loc is a type which rc.Params["legend.loc"] supports.
+
+    .. versionchanged:: 3.7
+
+    Parameters
+    ----------
+    loc : str | int | (float, float) | str((float, float))
+        The location of the legend.
+
+    Returns
+    -------
+    loc : str | int | (float, float) or raise ValueError exception
+        The location of the legend.
+    """
+    valid = {
+        "best",
+        "upper right", "upper left", "lower left", "lower right", "right",
+        "center left", "center right", "lower center", "upper center",
+        "center"
+    }
+
+    key = "legend.loc"
+    msg = (f"{loc!r} is not a valid value for {key}; supported values "
+           f"are {', '.join(valid)} | tuple[float, float] "
+           f"| str(tuple[float, float])")
+
+    if isinstance(loc, int):
+        return loc
+    if isinstance(loc, str):
+        loc = loc.lower()
+        if loc in valid:
+            return loc
+        if (loc.startswith('"') and loc.endswith('"')
+            or loc.startswith("'") and loc.endswith("'")) \
+                and loc[1:-1] in valid:
+            msg += "; remove quotes surrounding your string"
+        else:
+            loc = ast.literal_eval(loc)
+    if isinstance(loc, tuple):
+        if len(loc) == 2 \
+                and all(isinstance(e, numbers.Real)
+                        and float(e) >= 0.0 for e in loc):
```

Comment:
The value does not need to be positive here.  We can set negative numbers to have a legend outside of the axes:

```python
fig, ax = plt.subplots(layout="constrained")
ax.plot([0, 1], label="foo")
ax.legend(loc=[-0.3, 0.2])
plt.show()
```
![image](https://github.com/matplotlib/matplotlib/assets/10599679/3f69d164-08c2-4145-9bbe-195487340407)